server/microservices/profile/src/services/ProfileService.ts

```diff
@@ -0,0 +1,104 @@
+import jwt from 'jsonwebtoken';
+import ApiError from '../lib/ApiError';
+import IProfile from '../models/interfaces/Profile';
+import ProfileCreationBodyDTO from '../models/interfaces/ProfileCreationBodyDTO';
+import ProfileDTO from '../models/interfaces/ProfileDTO';
+import PendingProfile from '../models/sequelize/PendingProfile';
+import Profile from '../models/sequelize/Profile';
+import ProfileContact from '../models/sequelize/ProfileContacts';
+
+export default class ProfileService {
+  public async createProfile(body?: ProfileCreationBodyDTO): Promise<void> {
+    if (!body || !body.id || !body.email)
+      throw ApiError.badRequest('Missing id or email');
+    if (await PendingProfile.findOne({ where: { id: body.id } }))
+      throw ApiError.badRequest('Pending profile with this id already exists');
+    await PendingProfile.create({
+      id: body.id,
+      email: body.email,
+    });
+  }
+
+  public async confirmProfile(body?: Omit<ProfileDTO, 'id'>): Promise<void> {
+    if (!body) throw ApiError.badRequest('There is no body');
+    const pendingProfile = await PendingProfile.findOne({
+      where: { email: body.email },
+    });
+    if (!pendingProfile)
+      throw ApiError.badRequest('Pending profile with this id not found');
+    if (!body.firstName || !body.tag || !body.birthDate || !body.telephone)
+      throw ApiError.badRequest('Missing some fields');
+    await this.validateTag(body.tag);
+    await Profile.create({
+      id: pendingProfile.id,
+      email: pendingProfile.email,
+      firstName: body.firstName,
+      lastName: body.lastName,
+      tag: body.tag,
+      avatar: body.avatar,
+      birthDate: body.birthDate,
+      telephone: body.telephone,
+    });
+    await pendingProfile.destroy();
+  }
+
+  public async validateTag(tag: string): Promise<void> {
+    const profile = await Profile.findOne({ where: { tag } });
+    if (profile) throw ApiError.badRequest('Tag already exists');
+  }
+
+  public async getProfile(id: string): Promise<IProfile> {
+    const profile = await Profile.findOne({ where: { id } });
+    if (!profile) throw ApiError.notFound('Profile not found');
+    return profile;
```

Comment:
Returning raw database intances is bad (Nosov said so :) ). Probably better to use a dto

Reply:
Do you mean 
```
res.status(200).json(JSON.parse(JSON.stringify(profile)))
```
?

Reply:
It is already serialized by res.json() and has only what we need to send

Reply:
No, I meant we need to explicitly control what object is sent as a response (what fields etc.)
Something like
```suggestion
    return res.status(200).json(new ProfileResponseDTO(profile));
```
so for example utility fields that we don't need are not included in the response (e.g. __version in MongoDB)
Once again, this is what Nosov recommended

Reply:
I tried to implement this in chat service so take a look if you want

Reply:
Got it

Reply:
Fixed in 8efdae6